src/ProductBundle/DependencyInjection/SonataProductExtension.php

```diff
@@ -210,16 +209,6 @@ public function registerDoctrineMapping(array $config): void
              'orphanRemoval' => false,
         ]);
 
-        $collector->addAssociation($config['class']['collection'], 'mapOneToMany', [
-            'fieldName' => 'productCollection',
-            'targetEntity' => $config['class']['product_collection'],
-            'cascade' => [
-                'persist',
-            ],
-            'mappedBy' => 'collection',
-            'orphanRemoval' => false,
-        ]);
```

Comment:
Would it make sense to do this conditionally, if `property_exists($config['class']['collection'], 'productCollection')`? I did not follow the whole issue, but I since you say this fix might actually break people's stuff…

Reply:
Well, that is what I proposed, to add it conditionally. In theory, it could break if someone relies on the association being there right? 

I will update this PR to add it conditionally but propose to remove it altogether in the master branch.

Reply:
> I will update this PR to add it conditionally but propose to remove it altogether in the master branch.

`[this is the way](https://media1.tenor.com/images/614de537ea516c274f9c95e722fbf9d5/tenor.gif?itemid=15630699)